### PR TITLE
add growl message to display warning messages from response headers

### LINKED ISSUE #8502: Deploying a daemonset to the `default` namespace with an RKE2/K3s v1.25 cluster with PSACT `rancher-restricted` does not error out as expected
**Setup**
- Rancher version: `v2.7-head` https://github.com/rancher/rancher/commit/8835c43d5d3e1599695a5d9bba989afac5c93436
- Browser type & version: Chrome

**Describe the bug**

After provisioning an RKE2 v1.25 node provider cluster with PSACT set to `rancher-restricted`, I went to validate that creating workloads would fail. Attempting to create a test deployment with `nginx:latest` in the `default` namespace failed as expected:

image

I went to do the same for daemonsets, but I see the following:

image

Curious, I went to run `kubectl get <daemonset> -n default` to see if I would get an error in the event logs, which I did:

```
> kubectl describe daemonset.apps/test -n default
Name:           test
Selector:       workload.user.cattle.io/workloadselector=apps.daemonset-default-test
Node-Selector:  <none>
Labels:         workload.user.cattle.io/workloadselector=apps.daemonset-default-test
Annotations:    deprecated.daemonset.template.generation: 1
Desired Number of Nodes Scheduled: 1
Current Number of Nodes Scheduled: 0
Number of Nodes Scheduled with Up-to-date Pods: 0
Number of Nodes Scheduled with Available Pods: 0
Number of Nodes Misscheduled: 0
Pods Status:  0 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:  workload.user.cattle.io/workloadselector=apps.daemonset-default-test
  Containers:
   container-0:
    Image:        nginx:latest
    Port:         <none>
    Host Port:    <none>
    Environment:  <none>
    Mounts:       <none>
  Volumes:        <none>
Events:
  Type     Reason        Age                    From                  Message
  ----     ------        ----                   ----                  -------
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-hp4zz" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-czk74" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-lbktm" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-lp8qt" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-ncs2t" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-dvt59" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-2r5pn" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m23s                  daemonset-controller  Error creating: pods "test-266s8" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  5m22s                  daemonset-controller  Error creating: pods "test-xthkr" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
  Warning  FailedCreate  2m39s (x7 over 5m21s)  daemonset-controller  (combined from similar events): Error creating: pods "test-lbtst" is forbidden: violates PodSecurity "restricted:latest": unrestricted capabilities (container "container-0" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (container "container-0" must not set securityContext.runAsNonRoot=false), seccompProfile (pod or container "container-0" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```
As the logs indicate the expected failure, it is odd that the test daemonset does not reflect this error message outright, but only in the logs.

**To Reproduce**

1. As a standard user, provision a RKE2/K3s downstream node driver v1.25 cluster with PSACT set to `rancher-restricted`.
2. Validate that the cluster and nodes come up as `Active` in Cluster Management and Cluster Explorer.
3. Navigate to top-left menu > cluster > Workloads > Create.
4. Create a test daemonset with `nginx:latest` as the container image in the `default` namespace.
5. Note that it stays in `In Progress` state and does not error out as expected.
6. Click the `kubectl shell` and run command `kubectl describe <daemon set name> -n default`; validate the event logs show the expected error.

**Result**
The test daemonset is stuck in `In Progress` with no error message shown.

**Expected Result**

The test daemonset should error out and give the message shown in the logs, same as the test deployment did.

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #8502 

### Occurred changes and/or fixed issues

- add growl message to display warning messages from response headers
- update growl styling based on feedback from @edenhernandez-suse 

### Technical notes summary

- This was the only mechanism that we have of "easy access" to display warnings from response headers. Also, these types of warnings are non-persistent (they don't exist on the object resource), so we can't show them on the details screen (ex: in this particular case `DaemonSets`, but you have the related events).

### Areas or cases that should be tested

- Test scenario of the above mentioned issue #8502 to check that the mechanism works as expected

### Areas which could experience regressions

- Took the time creating a few different resource instances (clusters, global role, pod, persistent volume, some things in fleet, etc) just to check if there were any additional growls that we didn't know about (we don't want to flood the UI with these messages) and no other warning/growl was generated in consequence. (only the pod, because it's supposed to fail in the same scenario as per issue description)

### Screenshot/Video
**Growl**

Screenshot 2023-08-24 at 12 51 16

**DaemonSets details page - events table**
Screenshot 2023-08-23 at 11 11 28

### CODE REVIEW

**Review by richard-cox — CHANGES_REQUESTED** [review]
- Something that can be done as part of this PR, ensure that if opt.method === 'PUT' we make it to the `handleValidationWarnings` function and it's also handled when determining create / update. To reproduce this create / edit any resource via YAML
- Unrelated to this PR, YAML editing error messages are broken https://github.com/rancher/dashboard/issues/9592
- i've tried for a while to get any other type of warning header than validation and this PRs message. will continue with that tomorrow. Will also work out if there's a _super simple_ way to tweak the filter used to determine if the warning goes in a growl. We've had a lot of issues in the past where spurious growls bite us real hard, so would like a way for customers to either disable or configure it.

File: shell/plugins/steve/actions.js [hunk 1]
```diff
@@ -197,13 +197,34 @@ export default {
...
-    function handleValidationWarnings(res) {
+    function handleValidationWarnings(res, that, rootGetters, method) {
       const warnings = (res.headers?.warning || '').split(',');
 
       if (!warnings.length || !warnings[0]) {
         return;
       }
 
+      // let's filter out "unknown field" warnings from the response headers
+      const relevantWarnings = warnings.filter((w) => !w.includes('299 - unknown field')) || [];
```
[the remark this answers is not in the document]
Comment (richard-cox) [reply shell/plugins/steve/actions.js]: n/a

[the remark this answers is not in the document]
Comment (richard-cox) [reply shell/plugins/steve/actions.js]: n/a given new changes

**Review by richard-cox — APPROVED** [review]
I've updated a lot of things, more detail in the commit message. These all need a good review though, @aalves08 if you're happy with the changes this can go in.

**Comment by aalves08** [comment]
Uau @richard-cox , nice job with this PR! 👏 you went all out with this one! I've reviewed your code changes + tested the functionality and all works as expected. I'm going to merge this. Thank you very much for your contribution. 🙇

**Comment by aalves08** [comment]
Ah, you need to re-approve it @richard-cox ! I can't, as a PR owner, so that this can be merged. 🙏

**Review by richard-cox — APPROVED** [review]

**Pull request merged**